src/NuGet.Core/NuGet.Packaging/Licenses/NuGetLicenseExpressionExtensions.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Packaging
+{
+    public static class NuGetLicenseExpressionExtensions
+    {
+        /// <summary>
+        /// Determines whether all the licenses and exceptions are not deprecated.
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <returns></returns>
+        public static bool HasOnlyStandardIdentifiers(this NuGetLicenseExpression expression)
+        {
+            switch (expression.Type)
+            {
+                case NuGetLicenseExpressionType.License:
+
+                    return (expression as NuGetLicense).IsStandardLicense;
+
+                case NuGetLicenseExpressionType.Operator:
+
+                    var licenseOperator = expression as NuGetLicenseOperator;
+                    switch (licenseOperator.OperatorType)
+                    {
+                        case NuGetLicenseOperatorType.LogicalOperator:
+                            var logicalOperator = expression as NuGetLicenseLogicalOperator;
+                            return logicalOperator.Left.HasOnlyStandardIdentifiers() && logicalOperator.Right.HasOnlyStandardIdentifiers();
+                        case NuGetLicenseOperatorType.WithOperator:
+                            var withOperator = expression as NuGetLicenseWithOperator;
+                            return withOperator.License.IsStandardLicense;
+                        default:
+                            return false;
+                    }
+
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary>
+        /// A leaf node in an expression can only be a License or an Exception. Run a func on each one.
+        /// </summary>
+        /// <param name="expression"></param>
+        /// <param name="licenseProcessor"></param>
+        /// <param name="exceptionProcessor"></param>
+        public static void OnEachLeafNode(this NuGetLicenseExpression expression, Action<NuGetLicense> licenseProcessor, Action<NuGetLicenseException> exceptionProcessor)
```

Comment:
>OnEachLeafNode [](start = 27, length = 14)

Perhaps we should document this as pre-order traversal? That's what it is, right?

Reply:
Not sure we can call it either since we don't really act on the current node :)